pdf_generator/tex_generation/template_engine.py

```
@@ -0,0 +1,166 @@
+import logging
+from base64 import decodebytes
+from collections import OrderedDict
+from pathlib import Path
+from time import localtime, strftime
+
+import jinja2
+from common_helper_files import human_readable_file_size
+
+GENERIC_TEMPLATE = 'generic.tex'
+
+
+def byte_number_filter(number, verbose=True):
+    if isinstance(number, (int, float)):
+        if verbose:
+            return '{} ({})'.format(human_readable_file_size(int(number)), format(number, ',d') + ' Byte')
+        return human_readable_file_size(int(number))
+    return 'not available'
+
+
+def nice_unix_time(unix_time_stamp):
+    '''
+    input unix_time_stamp
+    output string 'YYYY-MM-DD HH:MM:SS'
+    '''
+    if isinstance(unix_time_stamp, (int, float)):
+        tmp = localtime(unix_time_stamp)
+        return strftime('%Y-%m-%d %H:%M:%S', tmp)
+    return 'not available'
+
+
+def nice_number_filter(number):
+    if isinstance(number, int):
+        return '{:,}'.format(number)
+    if isinstance(number, float):
+        return '{:,.2f}'.format(number)
+    if isinstance(number, str):
+        try:
+            return str(int(number))
+        except ValueError:
+            pass
```

better use `suppress()` instead of `except: pass`